### PR TITLE
Backport fastparquet 0.7 compat (PR #42954 and #42919)

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] xref #42954 and #42919
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

**This is a non-trivial backport. Please first review changed files before merging. Thanks**

### CODE REVIEW

**Comment by lithomas1** [comment]
What? How is CI picking up only fastparquet 0.7.0. This needs more investigation...

**Comment by lithomas1** [comment]
Anyone on the core team want to restart just the database builds? I just want to confirm that this is not flaky.

File: pandas/tests/io/test_parquet.py [hunk 1]
```diff
@@ -1049,9 +1063,14 @@ def test_timezone_aware_index(self, fp, timezone_aware_date_list):
...
 
-    def test_use_nullable_dtypes_not_supported(self, fp):
+    def test_use_nullable_dtypes_not_supported(self, monkeypatch, fp):
         df = pd.DataFrame({"a": [1, 2]})
 
+        # This is supported now in fastparquet 0.7.1 and above actually
+        # Still need to ensure that this raises in all versions below
+        import fastparquet as fp
+
+        monkeypatch.setattr(fp, "__version__", "0.4")
```
Comment (jorisvandenbossche) [inline pandas/tests/io/test_parquet.py:1073]: This monkeypatch actually shouldn't be needed?

**Comment by jorisvandenbossche** [comment]
Restarted, but it was indeed installing fastparquet 0.7.0 ...

File: pandas/tests/io/test_parquet.py (hunk 1 above)
Comment (lithomas1) [reply pandas/tests/io/test_parquet.py:1073]: Yep, forgot to revert this part. It's actually present in master too. Will do after this, so the bot can auto backport.

**Comment by lithomas1** [comment]
Ok figured it out. fastparquet decided to actually enforce numpy>1.18 in 0.7.1.

**Comment by jorisvandenbossche** [comment]
Thanks! (and indeed, a follow-up PR for my comment is the easiest ;))

**Pull request merged**